src/script/common/c_content.cpp

```diff
@@ -2048,8 +2048,9 @@ bool read_tree_def(lua_State *L, int idx, const NodeDefManager *ndef,
 }
 
 /******************************************************************************/
-#if defined(JSONCPP_STRING) || !(JSONCPP_VERSION_MAJOR < 1 || JSONCPP_VERSION_MINOR < 9)
-#define HAVE_JSON_STRING
+
+#if JSONCPP_VERSION_HEXA < 0x01000000 /* 1.0.0 */
+#warning "Your JsonCPP version is too old and might not be compatible."
```

Comment:
1.0.0 is from 2014 and the first version with c++11 support. there is no reason to be lenient.